### PR TITLE
Doc/issue156 readthedocs ii

### PR DESCRIPTION
**Pull request by gungula**
Reworked readthedocs and did some small further changes like in the man pages or the specifications.

### CODE REVIEW

File: docs/source/02_project_procedure/021_critical/021_ipc.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+| `D-Bus Wikipedia <https://en.wikipedia.org/wiki/D-Bus>`__ 
+| `D-Bus documentation Rust <https://docs.rs/dbus/latest/dbus/>`__ 
+| `D-Bus create from freedesktop.org <https://dbus.pages.freedesktop.org/zbus/>`__
+| `D-Bus interface for Rust <https://github.com/diwic/dbus-rs>`__ 
+
+- high level IPC mechanism 
+- based on unix sockets 
+- widely used in projects such as Gnome and KDE 
+- offers message queing, tow way communication and is supposed to offer a easy to use interface
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_ipc.rst:102]: message queuing

File: docs/source/02_project_procedure/021_critical/021_ipc.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+IPC mechanism.
+
+Implementation
+~~~~~~~~~~~~~~
+
+Available libraries
+^^^^^^^^^^^^^^^^^^^
+
+There seem to be two main projects striving to provide D-Bus support for
+rust.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_ipc.rst:137]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_ipc.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+| `Zbus project repository <https://gitlab.freedesktop.org/dbus/zbus/-/tree/main>`__
+| `Zbus crate <https://crates.io/crates/zbus>`__
+| `Zbus documentation <https://dbus.pages.freedesktop.org/zbus/>`__ 
+
+- official D-Bus rust implementation by the freedesktop.org foundation
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_ipc.rst:143]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_ipc.rst [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+- wrapper library for libdbus -> libdbus dependency - examples
+
+.. _conclusion-1:
+
+Conclusion
+^^^^^^^^^^
+
+Zbus seems to have some advantages over D-Bus-rs, mainly: - official
+freedesktop.org library - pure rust -> no libdbus dependency -
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_ipc.rst:159]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+The window manager should at all stages be reactive and only use a
+minimal amount of resources. In order to assure this polling of threads
+should never be implemented.
+
+Event Sources
+-------------
+
+Events for the window manager can be created by multiple sources.
+
+1. ``X-Events`` Generated by the X-Server and regarding userinput,
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:16]: user input

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+   clientsrequests and more.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:17]: clients requests

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+In order to be able to handle these two event sources two possible
+solutions are available:
+
+1. Polling of both event-sources
+
+| poll for X-Event
+| poll for IPC-event
+| execute received event
+
+This layout allows for a very simple single threaded implementation of
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:30]: allows a very...

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+the window manager but requires a frequent polling loop in order to
+allow for a reactive behavior.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:32]: allow a ...

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+Therefore this layout is not suitable to our requirements as this would
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:34]: suitable for

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+**=> Therefore we will implement this layout.**
+
+Technical solution
+------------------
+
+Waiting for event
+~~~~~~~~~~~~~~~~~
+
+The following documents functions of the chosen libraries that support
+waiting on an event:
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:65]: waiting for

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+1. X-Event The ``x11rb`` library supports waiting on an event:
+   `wait_for_event
+   method <https://smithay.github.io/smithay/x11rb/connection/trait.Connection.html#method.wait_for_event>`__
+
+2. IPC-Event Zbus is asynchronous in nature. The handling functions are
+   registered to the server and executed when an event is received.
+   `Zbus Example <https://docs.rs/zbus/latest/zbus/>`__
+
+3. mainloop As shared queue the channel implementation of the rust
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_reactiveness.rst:75]: mainloops as

File: docs/source/02_project_procedure/021_critical/021_rust.rst [hunk 12]
```diff
@@ -0,0 +1,25 @@
+.. _rust:
+
+===============================
+Rust as implementation language
+===============================
+
+This document outlines why rust has been chosen as the language of
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_rust.rst:7]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_rust.rst [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+implementation for this project. But it needs to be mentioned that this
+choice was not purely made out of technical reasoning, because we think
+rust is fun and we enjoy to learn this language.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_rust.rst:10]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_rust.rst [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+Technical arguments for Rust
+----------------------------
+
+Rust is a good choice for a system level language as it allows direct
+memory acces but also object orientation and offers a memory safe
+environment. Further this is achieved without a resource intensive
+garbage collection and high speeds in the realm of C are possible. Even
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_rust.rst:18]: ~are possible~

File: docs/source/02_project_procedure/021_critical/021_rust.rst [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+though rust is still “relatively” new, it already has a stable ecosystem
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_rust.rst:19]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_rust.rst [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+and a lot of libraries supporting it.
+
+A first search for rust libraries allowing connections to the X-Server
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_rust.rst:22]: Rust - names are written with capital letters

File: docs/source/02_project_procedure/021_critical/021_xcb.rst [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+-------------------------------
+
+-  Toolkit implementation
+-  Direct protocol programming (see section *Why not to use XCB*)
+-  Leightweight emulation of commonly used portions of the Xlib API
+-  XCB does not lock itself while waiting for a response it send to a
+   x-server instance like Xlib. This avoids needless stalling while a
+   request is processed. Instead, XCB binds a cookie to a request which
+   can then be used to ask for a pointer to the corresponding reply.
+   This not only enables reading of the reply only when it is required,
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_xcb.rst:35]: this sentence sounds weird to me

File: docs/source/02_project_procedure/021_critical/021_xcb.rst [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+   WWWW--RRRR
+
+The total time is
+``N * T_write + max(0, T_round_trip - (N-1) * T_write) + N * T_read``.
+
+Conclusion
+~~~~~~~~~~
+
+XCB offers considerably faster event handling The tutorial linked at the
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_xcb.rst:74]: handling.

File: docs/source/02_project_procedure/021_critical/021_xcb.rst [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+top of this document also proiveds the source code and results of a
+benchmarking which lead to the same results.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_xcb.rst:76]: leads - im not sure tho

File: docs/source/02_project_procedure/021_critical/021_xcb.rst [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+
+Why to not use Xlib
+-------------------
+
+**Xlib is quite big:** Xlib is bigger then XCB and can therefor not be
+used with minimalistic systems. However, the target groups of this
+project are (semi-)well equiped computers, which makes this not as much
+of an advantage.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_xcb.rst:84]: disadvantage?

File: docs/source/02_project_procedure/021_critical/021_xcb.rst [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+some degree work with multiple threads too but its API was not designed
+for this purpose which makes it difficult to work with as well as
+error-prone.
+
+Why not to use XCB
+------------------
+
+**Direct protocol access:** This can be good or not depending on the
+system an application will run on. Xlib performs chaching, layering and
+other optimizations by itself. XCB does not provide this feature.
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/021_critical/021_xcb.rst:100]: maybe:  _these features_

File: docs/source/02_project_procedure/026_project_handbook.rst [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+Every merge into the ``main`` branch has to be accepted and reviewed
+through a pull request. There should not be any rebase onto ``main``.
+Working methods on the feature branches are open to developer.
+
+Testing
+-------
+
+All test logs are to be stored in the subdirectory ``test_logs``. Those
+will not be published on GitHub. Upcoming issues should be documented
+with Git-issues with the following format:
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/026_project_handbook.rst:59]: Git-issue or Git issue?

File: docs/source/02_project_procedure/026_project_handbook.rst [hunk 23]
```diff
@@ -0,0 +1,110 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+All research results and discussions will be stored in the ``concepts``
+folder. All documents will be written in markdown - there are no other
+formal restrictions.
+
+Ticketing
+~~~~~~~~~
+
+Every task will be documented with a Git issue. The current status will
```
Comment (Philipp6802) [inline docs/source/02_project_procedure/026_project_handbook.rst:20]: Git-issue or Git issue?

**Review by Philipp6802 — CHANGES_REQUESTED** [review]

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst (hunk 6 above)
Comment (Philipp6802) [reply docs/source/02_project_procedure/021_critical/021_reactiveness.rst:17]: not sure tho

File: docs/source/02_project_procedure/021_critical/021_reactiveness.rst (hunk 5 above)
Comment (Philipp6802) [reply docs/source/02_project_procedure/021_critical/021_reactiveness.rst:16]: not sure tho

File: docs/source/02_project_procedure/026_project_handbook.rst (hunk 23 above)
Comment (gungula) [reply docs/source/02_project_procedure/026_project_handbook.rst:20]: git issue

File: docs/source/02_project_procedure/021_critical/021_rust.rst (hunk 14 above)
Comment (gungula) [reply docs/source/02_project_procedure/021_critical/021_rust.rst:18]: The sentence would be incomplete?!

File: docs/source/02_project_procedure/021_critical/021_xcb.rst (hunk 21 above)
Comment (gungula) [reply docs/source/02_project_procedure/021_critical/021_xcb.rst:100]: No, it is the feature of direct protocol access

**Review by Philipp6802 — APPROVED** [review]

**Pull request merged**